at_client/lib/src/key_stream/key_stream_mixin.dart

```diff
@@ -0,0 +1,419 @@
+import 'dart:async';
+
+import 'package:at_client/at_client.dart' show AtClient, AtClientManager, AtNotification;
+import 'package:at_client/src/listener/at_sign_change_listener.dart';
+import 'package:at_client/src/listener/switch_at_sign_event.dart';
+import 'package:at_commons/at_commons.dart' show AtKey, AtValue;
+import 'package:at_utils/at_logger.dart';
+
+import 'package:meta/meta.dart';
+
+abstract class KeyStreamMixin<T> implements Stream<T> {
+  /// A subscription that listens to notifications.
+  @visibleForTesting
+  late StreamSubscription<AtNotification> notificationSubscription;
+
+  late AtClientManager _atClientManager;
+
+  static final AtSignLogger _logger = AtSignLogger('KeyStream');
+
+  @visibleForTesting
+  static bool disposeOnAtsignChange = true;
```

Comment:
@gkc there was in fact was an instance where we switch atsigns. I've added this flag to check whether we should listen to the atsign change.

Reply:
👍👍👍